fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/UserRolesSyncer.java

```diff
@@ -201,6 +201,41 @@ public long syncAndReturn(List<String> roles) {
     }
   }
 
+  public long syncServiceAccounts(String serviceAccountId, List<String> roles) {
+    // So that fiat will pull fresh permissions
+    permissionsResolver.clearCache();
+
+    // Store this service account as a user with associated resources via `resolveAndMerge`.
+    List<Role> convertedRoles =
+        roles.stream()
+            .map(extRole -> new Role().setSource(Role.Source.EXTERNAL).setName(extRole))
```

Comment:
One thing I've not dug, though I THINK it's ok for Fiat.  Just making sure this is if this is how normal service accounts are handled.  E.g. there is some sync behavior on service accounts to try to avoid calling API's of remove systems (when using google/github for auth or ldap).

Reply:
I think so - it seems to be done here as a part of the normal sync as well https://github.com/spinnaker/fiat/commit/906ecd7724441bc4b4b4aac349306fa3eded9359#diff-9cc5e46be2367bea393fc1b1449ef306c0c9f6be2802833a267b644cdb4c7b4cR172